### PR TITLE
Chris tests: SignUp & CallItEven

### PR DESCRIPTION
**Pull request by chrisperry-sd**

### CODE REVIEW

File: client/src/__tests__/callItEven.test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      setTransactions={val => {
+        console.log('transaction', val);
+      }}
+      isLoading={false}
+      setIsLoading={val => {
+        console.log('isloading', val);
+      }} />);
+    const a = await screen.getByText('Wipe that debt');
+    userEvent.click(a);
+    expect(navigate).toHaveBeenCalledTimes(1);
```
Comment (ilyadusoleil) [inline client/src/__tests__/callItEven.test.js:50]: Could also test to check for inputs

File: client/src/__tests__/callItEven.test.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+describe('Call it even', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  })
+  it('displays correct summary and users partner', async () => {
+    render(<CallItEven users={users} currency={currency} summary={summary} />);
+    expect(screen.getByText(/M\s*owes you\s*£5/)).toBeInTheDocument();
+  })
+  //mock form input - check (wipe that debt) naviagtes away from page
```
Comment (ilyadusoleil) [inline client/src/__tests__/callItEven.test.js]: *navigates

File: client/src/__tests__/callItEven.test.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      summary={summary}
+      setTransactions={val => {
+        console.log('transaction', val);
+      }}
+      isLoading={false}
+      setIsLoading={val => {
+        console.log('isloading', val);
+      }} />);
+
+    const LeaveNote = screen.getByRole('LeaveNote');
```
Comment (ilyadusoleil) [inline client/src/__tests__/callItEven.test.js:66]: You could use .getByRole('textbox', {name: NAME})
You'd have to add an aria label, but I'd argue at least that's useful for accessibility purposes

File: client/src/__tests__/callItEven.test.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    expect(LeaveNote).toBeInTheDocument();
+
+    const a = screen.getByText('Wipe that debt');
```
Comment (ilyadusoleil) [inline client/src/__tests__/callItEven.test.js]: var name

File: client/src/pages/CallItEven.js [hunk 5]
```diff
@@ -71,6 +70,7 @@ const CallItEven = ({
         <FormSection>
           <FormLabel htmlFor="wipe-description">Leave a note:</FormLabel>
           <FormInput
+            role="LeaveNote"
```
Comment (ilyadusoleil) [inline client/src/pages/CallItEven.js:73]: or aria-label

File: client/src/pages/Dashboard.js [hunk 6]
```diff
@@ -54,7 +54,7 @@ export default function Dashboard({
                   width: 100%;
                 `}
               >
-                {/* <Lottie options={celebrate} /> */}
+                <Lottie options={celebrate} />
```
Comment (ilyadusoleil) [inline client/src/pages/Dashboard.js]: thanks :)

File: client/src/__tests__/inspect.test.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
+import userEvent from '@testing-library/user-event'
+import { navigate } from '@reach/router';
+import '@testing-library/jest-dom/extend-expect'
+import Inspect from '../pages/Inspect';
+import { deleteTransaction, getOneTransaction, getTransactions, editTransaction } from '../services/TransactionService';
+
+jest.mock('../services/TransactionService');
+getOneTransaction.mockResolvedValue('upload');
+getTransactions.mockResolvedValue('upload');
```
Comment (ilyadusoleil) [inline client/src/__tests__/inspect.test.js:11]: I'm guessing you aren't using this mock resolved value, but perhaps don't use the word 'upload' for when you are doing a get request

File: client/src/__tests__/inspect.test.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+const users =
+{
+  _id: "5f75e077dd559746a01dda55",
+  lead: "C",
+  leadEmail: "1234@test",
+  partner: "M",
+  partnerEmail: "123@test"
+};
+const currency = '£';
+// _id
```
Comment (ilyadusoleil) [inline client/src/__tests__/inspect.test.js]: //ehh commented code
Though i've probably left a bunch of commented code in my pull requests too

File: client/src/__tests__/inspect.test.js [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+  afterEach(() => {
+    jest.clearAllMocks();
+  })
+  it('"Save my edits" directs away from page', async () => {
+    render(<Inspect
+      users={users}
+      setTransactions={val => { console.log('transaction', val) }}
+      setIsLoading={val => { console.log('isloading', val) }} />)
+
+    const a = screen.getByText('Save my edits');
```
Comment (ilyadusoleil) [inline client/src/__tests__/inspect.test.js]: var name

File: client/src/__tests__/inspect.test.js [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+    expect(a).toBeInTheDocument();
+
+    userEvent.click(a);
+    expect(navigate).toHaveBeenCalledTimes(1);
```
Comment (ilyadusoleil) [inline client/src/__tests__/inspect.test.js:43]: again, could also check input to navigate function

File: client/src/pages/Inspect.js [hunk 11]
```diff
@@ -89,35 +89,38 @@ const Inspect = ({ _id, users, currency, setTransactions, setIsLoading }) => {
...
             type="text"
             name="bill-item"
+            aria-label="bill-item"
             onChange={event => setItem(event.target.value)}
             value={item}
             required
           />
         </FormSection>
-        <FormSection>
+        <FormSection role="bill-amount">
```
Comment (ilyadusoleil) [inline client/src/pages/Inspect.js:103]: Are you using these roles?

File: client/src/pages/Inspect.js [hunk 12]
```diff
@@ -161,7 +167,7 @@ const Inspect = ({ _id, users, currency, setTransactions, setIsLoading }) => {
       </form>
       <form onSubmit={deleteMe}>
         <FormSection>
-          <DeleteButton type="submit">
+          <DeleteButton type="submit" role="deleteTrans">
```
Comment (ilyadusoleil) [inline client/src/pages/Inspect.js:170]: shouldn't need a role here - just use role: button

File: client/src/pages/SignUp.js [hunk 13]
```diff
@@ -124,12 +128,13 @@ const SignUp = ({ setIsAuthenticated, setIsLoading }) => {
...
             type="text"
             name="partnerEmail"
+            aria-label="partners-email"
             value={state.partnerEmail}
             onChange={handleChange}
             required
           />
         </FormSection>
-        <FormButton type="submit">Let&apos;s get started</FormButton>
+        <FormButton role="button" type="submit">Let&apos;s get started</FormButton>
```
Comment (ilyadusoleil) [inline client/src/pages/SignUp.js:137]: doesn't this not have this role by default?

File: client/src/__tests__/history.test.js [hunk 14]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { render, screen, cleanup, act, waitFor } from '@testing-library/react';
```
Comment (ilyadusoleil) [inline client/src/__tests__/history.test.js:2]: 👍

File: client/src/__tests__/signUp.test.js [hunk 15]
```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
```
Comment (ilyadusoleil) [inline client/src/__tests__/signUp.test.js]: DOn't need to include fireEvent here

File: client/src/__tests__/signUp.test.js [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+import userEvent from '@testing-library/user-event'
+import { navigate } from '@reach/router';
+import '@testing-library/jest-dom/extend-expect'
+import SignUp, {localStorage} from '../pages/SignUp';
```
Comment (ilyadusoleil) [inline client/src/__tests__/signUp.test.js]: HMm there is a mock for localStorage npm package that i've included in one of my changes. If you want you coudl look at updating this test to use that instead (but will also be fine without it)

Comment (chrisperry-sd) [reply client/src/__tests__/signUp.test.js]: Just seen your test for this in the login test. Will implement it here 🤙

File: client/src/__tests__/signUp.test.js [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+    expect(partnerName).toHaveValue('Tamara')
+
+
+    const partnerEmail = screen.getByLabelText('partners-email');
+    expect(partnerEmail).toBeInTheDocument();
+    userEvent.type(partnerEmail, 'tamara@test');
+    expect(partnerEmail).toHaveValue('tamara@test')
+
+
+    const button = screen.getByRole('button');
```
Comment (ilyadusoleil) [inline client/src/__tests__/signUp.test.js:60]: perhaps signUpButton to be more descriptive

File: client/src/__tests__/signUp.test.js [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+    expect(button).toBeInTheDocument();
+    userEvent.click(button);
+
+    await waitFor(() => {
```
Comment (ilyadusoleil) [inline client/src/__tests__/signUp.test.js:64]: does this work without the await here?

**Review by ilyadusoleil — COMMENTED** [review]
I don't really care, but might be worth looking at line spacing (sometimes you have no space where I'd personally put a space, and sometimes there is double spacing [probably shouldn't have that]), and make sure you use prettier on your files before uploading too 😀

**Pull request merged**